hazelcast/proxy/multi_map.py

```diff
@@ -445,6 +449,47 @@ def put(self, key: KeyType, value: ValueType) -> Future[bool]:
         request = multi_map_put_codec.encode_request(self.name, key_data, value_data, thread_id())
         return self._invoke_on_key(request, key_data, multi_map_put_codec.decode_response)
 
+    def put_all(self, multimap: typing.Dict[KeyType, typing.Sequence[ValueType]]) -> Future[None]:
+        """Stores the given Map in the MultiMap.
+
+        The results of concurrently mutating the given map are undefined.
+        No atomicity guarantees are given. It could be that in case of failure some of the key/value-pairs get written, while others are not.
+
+        Warning:
+            This method uses ``__hash__`` and ``__eq__`` methods of binary form
+            of the key, not the actual implementations of ``__hash__`` and
+            ``__eq__`` defined in key's class.
+
+        Args:
+            multimap: the map corresponds to multimap entries.
+        """
+        check_not_none(multimap, "multimap can't be None")
+        if not multimap:
+            return ImmediateFuture(None)
+
+        partition_service = self._context.partition_service
+        partition_map: typing.DefaultDict[int, typing.List[typing.Tuple[Data, typing.List[Data]]]] = defaultdict(list)
+
+        for key, values in multimap.items():
+            try:
+                check_not_none(values, "values can't be None")
```

Comment:
Java also does a null check for the key. I believe we can do that as well, because None is a valid dict key in Python